source/ast/symbols/MemberSymbols.cpp

```diff
@@ -766,13 +767,14 @@ void PrimitivePortSymbol::serializeTo(ASTSerializer& serializer) const {
 // Each 'bit' is one input value in the row.
 class BitTrie {
 public:
-    // Tries to insert the row. Returns nullptr if the row is
-    // successfully inserted, and a pointer to the previously
-    // inserted row with the same bit pattern if a collision
+    // Tries to insert the row. Returns nullopt if the row is
+    // successfully inserted, and a set of the previously
+    // inserted rows with the same bit pattern if a collision
     // is found.
     template<typename TAllocator>
-    const UdpEntrySyntax* insert(const UdpEntrySyntax& syntax, std::span<const char> inputs,
-                                 char stateChar, TAllocator& allocator) {
+    std::optional<std::set<const UdpEntrySyntax*>> insert(const UdpEntrySyntax& syntax,
```

Comment:
Rather than returning a std::set, which requires allocations (and you're not even specifically using the properties of a set as far as I can tell) can you make this function take a SmallVector that you add the matches to? Then it can return a boolean for whether the insertion was valid or not and the SmallVector can live on the stack in the caller.

Reply:
thanks, fixed